### PR TITLE
refactor allowance fetching to react-query

### PR DESCRIPTION
**Pull request by rolandpo**
allowances part of issue #960
refactoring allowance fetching in the style of #965

### CODE REVIEW

**Comment by rolandpo** [comment]
added APY fetching

File: dapp/src/services/apy.service.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (apyDayOptions.includes(days)) {
+          endpoint = `${process.env.APR_ANALYTICS_ENDPOINT}/${days}`
+          varName = `apy${days}`
+        } else {
+          throw new Error(`Unexpected days param: ${days}`)
+        }
+
+        try {
+          const response = await fetch(endpoint)
+          if (response.ok) {
```
Comment (sparrowDom) [inline dapp/src/services/apy.service.js:17]: If response is not ok react-query function should throw an error: 
https://react-query.tanstack.com/guides/query-functions#handling-and-throwing-errors

This way react query knows to attempt to refetch the data

File: dapp/src/services/apy.service.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            const json = await response.json()
+            return json.apy / 100
+          }
+        } catch (err) {
```
Comment (sparrowDom) [inline dapp/src/services/apy.service.js:21]: I would remove the whole try/catch statement. (and throw a new error when `!response.ok` as mentioned above.

**Review by sparrowDom — COMMENTED** [review]
Looks good aside from a small fix required

**Comment by rolandpo** [comment]
that should be fixed now

**Review by sparrowDom — APPROVED** [review]
Works nicely thanks 👍

**Pull request merged**